sysutils/puppet-agent/src/opnsense/service/conf/actions.d/actions_puppetagent.conf

```diff
@@ -0,0 +1,23 @@
+[start]
+command:/usr/local/etc/rc.d/puppet start
+parameters:
+type:script
+message:starting puppet agent
+
+[stop]
+command:/usr/local/etc/rc.d/puppet onestop
```

Comment:
```suggestion
command:/usr/local/etc/rc.d/puppet stop
```

Reply:
Here `onestop` is used to ensure Puppet Agent can be stopped when disabling it in the GUI. The usual flow in the service controller is as follows: First the template is written, which updates the rc.conf file and sets `puppet_enable="NO"`, hence the regular `stop` command would not be executed.

Is it advisable to change this flow? IMHO enforcing the stop action is not a bad thing, after all.

Reply:
If you want to stop a service you should not reconfigure it beforehand to leave the current / running configuration in place. I think the mutable controller does this.

Reply:
Ah yes, indeed, MutableServiceController uses a different flow. We'll migrate to it before releasing version 1.0 of the plugin. For the dev version I'd like to keep it so that we can start further testing.